classes-recipe/platsch.bbclass

```diff
@@ -0,0 +1,128 @@
+do_compile[cleandirs] = "${B}"
+
+platsch_do_compile() {
+    for IMG in ${SPLASH_IMAGES}; do
+        # filename has to be <basename>-<width>x<height>.<extension>
+        echo $IMG | sed -E -e "s/(.*)-(.*)x(.*)\.(.*)/\2 \3 \4/" |
+        {
+            read WIDTH HEIGHT EXT
+            if [ -z "${WIDTH}" -o -z "${HEIGHT}" -o -z "${EXT}" ]; then
+                bbfatal "platsch.bbclass: filename \"${IMG}\" does not satisfy scheme <basename>-<width>x<height>.<extension>."
+            fi
+
+            env MAGICK_CONFIGURE_PATH=${RECIPE_SYSROOT_NATIVE}/etc/ImageMagick-7 identify.im7 \
+                -format "%w %h\n" ${WORKDIR}/${IMG} |
+            {
+                read MGK_WIDTH MGK_HEIGHT
+                if [ "${WIDTH}" != "${MGK_WIDTH}" -o "${HEIGHT}" != "${MGK_HEIGHT}" ]; then
+                    bbfatal "platsch.bbclass: geometry encoded in filename \"${IMG}\" does not match geometry ${MGK_WIDTH}x${MGK_HEIGHT} found in file contents."
+                fi
+            }
+
+            BASENAME=$(basename --suffix=".${EXT}" ${IMG})
+            if [ -z "${BASENAME}" ]; then
+                bbfatal "platsch.bbclass: filename \"${IMG}\" does not satisfy scheme <basename>-<width>x<height>.<extension>."
+            fi
+
+            case ${SPLASH_FORMAT} in
+            RGB565)
+                # HACK: a quite compact way to create a bitmap in the RGB565 format is to
+                # let ImageMagick convert the source into its known BMP subtype RGB565 and
+                # to strip the BMP header by only using the final width * height * (5+6+5)/8
+                # bytes. For the i.MX6/i.MX8 the bitmap needs to be flipped (see [1]). This
+                # is done via the bitbake parameter MGK_CUSTOM_OPTION (see [2]). As the BMP
+                # format stores potential color profiles behind the desired pixel array at
+                # the end of the file those particularly need to be dropped before by the
+                # option "-strip" to let the hack succeed (see [3]). Explicitly request an
+                # RGB format to avoid color map and alpha channel (via asking for the
+                # TrueColor type, see [4]).
+                #
+                # [1] https://github.com/pengutronix/meta-ptx/pull/117#issuecomment-1474786927
+                # [2] https://imagemagick.org/script/command-line-options.php#flip
+                # [3] https://imagemagick.org/script/command-line-options.php#strip
+                # [4] https://imagemagick.org/script/command-line-options.php#type
+                #
+                RGB565_SIZE=$(expr ${WIDTH} "*" ${HEIGHT} "*" "(" 5 + 6 + 5 ")" / 8)
+                env MAGICK_CONFIGURE_PATH=${RECIPE_SYSROOT_NATIVE}/etc/ImageMagick-7 magick.im7 \
+                    ${WORKDIR}/${IMG} ${MGK_CUSTOM_OPTION} \
+                    -strip -type TrueColor -define bmp:subtype=${SPLASH_FORMAT} bmp:- | \
+                    tail -c ${RGB565_SIZE} > ${B}/${BASENAME}-${SPLASH_FORMAT}.bin
```

Comment:
Maybe check that the resulting file is really bigger than `${RGB565_SIZE}`?

Reply:
As exactly that (previously manual) check originally helped me to add ``ImageMagick``'s option ``-type TrueColor``, I now added a corresponding sanity check.